java/memory/src/main/java/io/netty/buffer/ArrowBuf.java

```diff
@@ -499,461 +225,212 @@ public boolean equals(Object obj) {
     return Float.intBitsToFloat(getInt(index));
   }
 
-  /**
-   * Gets a 64-bit long integer at the specified absolute {@code index} in
-   * this buffer in Big Endian Byte Order.
-   */
-  @Override
-  public long getLongLE(int index) {
-    chk(index, 8);
-    final long v = PlatformDependent.getLong(addr(index));
-    return Long.reverseBytes(v);
+  public ArrowBuf setFloat(int index, float value) {
+    chk(index, 4);
+    PlatformDependent.putInt(addr(index), Float.floatToRawIntBits(value));
+    return this;
   }
 
-  @Override
   public double getDouble(int index) {
     return Double.longBitsToDouble(getLong(index));
   }
 
-  @Override
-  public char getChar(int index) {
-    return (char) getShort(index);
-  }
-
-  @Override
-  public long getUnsignedInt(int index) {
-    return getInt(index) & 0xFFFFFFFFL;
-  }
-
-  @Override
-  public int getInt(int index) {
-    chk(index, 4);
-    final int v = PlatformDependent.getInt(addr(index));
-    return v;
-  }
-
-  /**
-   * Gets a 32-bit integer at the specified absolute {@code index} in
-   * this buffer in Big Endian Byte Order.
-   */
-  @Override
-  public int getIntLE(int index) {
-    chk(index, 4);
-    final int v = PlatformDependent.getInt(addr(index));
-    return Integer.reverseBytes(v);
-  }
-
-  @Override
-  public int getUnsignedShort(int index) {
-    return getShort(index) & 0xFFFF;
-  }
-
-  @Override
-  public short getShort(int index) {
-    chk(index, 2);
-    final short v = PlatformDependent.getShort(addr(index));
-    return v;
-  }
-
-  /**
-   * Gets a 16-bit short integer at the specified absolute {@code index} in
-   * this buffer in Big Endian Byte Order.
-   */
-  @Override
-  public short getShortLE(int index) {
-    final short v = PlatformDependent.getShort(addr(index));
-    return Short.reverseBytes(v);
-  }
-
-  /**
-   * Gets an unsigned 24-bit medium integer at the specified absolute
-   * {@code index} in this buffer.
-   */
-  @Override
-  public int getUnsignedMedium(int index) {
-    chk(index, 3);
-    final long addr = addr(index);
-    return (PlatformDependent.getByte(addr) & 0xff) << 16 |
-        (PlatformDependent.getShort(addr + 1) & 0xffff);
+  public ArrowBuf setDouble(int index, double value) {
+    chk(index, 8);
+    PlatformDependent.putLong(addr(index), Double.doubleToRawLongBits(value));
+    return this;
   }
 
-  /**
-   * Gets an unsigned 24-bit medium integer at the specified absolute {@code index} in
-   * this buffer in Big Endian Byte Order.
-   */
-  @Override
-  public int getUnsignedMediumLE(int index) {
-    chk(index, 3);
-    final long addr = addr(index);
-    return (PlatformDependent.getByte(addr) & 0xff) |
-        (Short.reverseBytes(PlatformDependent.getShort(addr + 1)) & 0xffff) << 8;
+  public char getChar(int index) {
+    return (char) getShort(index);
   }
 
-  @Override
-  public ArrowBuf setShort(int index, int value) {
+  public ArrowBuf setChar(int index, int value) {
```

Comment:
can we change this to either take a char, or replicate documentation saying which bits are ignored.

Reply:
Actually, is there a use-case to store 2-byte characters in Arrow (I'm not thinking of one off the top of my head).